sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -1598,11 +1598,38 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
             case b: HashJoin => b.leftKeys.head
           }.head.isInstanceOf[CollationKey])
         }
+
+        // Disable broadcast join to force sort merge join.
+        withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1") {
```

Comment:
would it maybe be easier to run the tests with default and -1 values of the conf, and then just assert that different joins are used based on the conf's value?

Reply:
how about
```
Seq("-1", "1").foreach(val =>
  withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> val) {
    ...
```